### PR TITLE
Fix JWT Signing from super tenant

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

**Comment by tharindu1st** [comment]
TestCase https://github.com/wso2/product-apim/pull/7100

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java [hunk 1]
```diff
@@ -161,9 +160,62 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
...
+        claims.put("consumerKey", jwtTokenInfoDTO.getConsumerKey());
         claims.put("backendJwt", jwtTokenInfoDTO.getBackendJwt());
 
         return claims;
     }
+
+    public byte[] signJWT(String assertion) throws APIManagementException {
+
+        PrivateKey privateKey;
+        KeyStoreManager tenantKSM = KeyStoreManager.getInstance(MultitenantConstants.SUPER_TENANT_ID);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java]: let's rename the variable to `superTenantKSM`

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java [hunk 2]
```diff
@@ -161,9 +160,62 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
...
+
+    /**
+     * Helper method to add public certificate to JWT_HEADER to signature verification.
+     *
+     * @throws APIManagementException
+     */
+    protected String addCertToHeader() throws APIManagementException {
+
+        try {
+            KeyStoreManager tenantKSM = KeyStoreManager.getInstance(MultitenantConstants.SUPER_TENANT_ID);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java]: same here.

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java [hunk 3]
```diff
@@ -161,9 +160,62 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
...
+            jwtHeader = jwtHeaderBuilder.toString();
+
+        } else if (SHA256_WITH_RSA.equals(signatureAlgorithm)) {
+            jwtHeader = addCertToHeader();
+        }
+        return jwtHeader;
+    }
+
+    /**
+     * Helper method to add public certificate to JWT_HEADER to signature verification.
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java]: Why we adding public certificate to the header? Correctly highligh what we adding to the header.

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java [hunk 4]
```diff
@@ -161,9 +160,62 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
...
     }
+
+    public byte[] signJWT(String assertion) throws APIManagementException {
+
+        PrivateKey privateKey;
+        KeyStoreManager superTenantKSM = KeyStoreManager.getInstance(MultitenantConstants.SUPER_TENANT_ID);
+        try {
+            privateKey = superTenantKSM.getDefaultPrivateKey();
+        } catch (Exception e) {
+            String msg = "Error while obtaining private key for super tenant";
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java]: log with msg? no need to duplicate strings

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/apikey/ApiKeyAuthenticator.java [hunk 5]
```diff
@@ -57,6 +57,8 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import javax.cache.Cache;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/apikey/ApiKeyAuthenticator.java]: why thise two added?

**Pull request merged**